# Party recommendations

In [1]:
import pandas as pd

In [2]:
recom = pd.read_csv("recommendations.csv")

In [3]:
recom

ref_id            date     party recommendation
0       224  7 février 1971       PRD            oui
1       224  7 février 1971       PDC            oui
2       224  7 février 1971        PS            oui
3       224  7 février 1971       UDC            oui
4       224  7 février 1971       PLS            oui
...     ...             ...       ...            ...
6231   6660     3 mars 2024       PEV            non
6232   6660     3 mars 2024       PVL            non
6233   6660     3 mars 2024  VERT-E-S            non
6234   6660     3 mars 2024       PST           none
6235   6660     3 mars 2024       UDF            oui

[6236 rows x 4 columns]

In [4]:
recom.describe()

ref_id         date party recommendation
count    6236         6236  6236           6236
unique    455          155    22              5
top       224  18 mai 2003    PS            oui
freq       16          117   455           2813

We can see that there are 455 referenda in the dataset, from 1971 to 2024. 22 Parties

In [5]:
byParty = {party: recom[recom["party"] == party] for party in recom["party"].unique()}

In [6]:
possibleRecoms = ["oui", "non", "blanc", "liberté de vote", "none"]
for party, df in byParty.items():
    print(party, "gave a vote recommendation for", len(df[df["recommendation"] != "none"]), "referenda")
    print(", ".join(r + ": " + str(len(df[df["recommendation"] == r])) for r in possibleRecoms))
    print()

PRD gave a vote recommendation for 316 referenda
oui: 191, non: 124, blanc: 0, liberté de vote: 1, none: 10

PDC gave a vote recommendation for 415 referenda
oui: 248, non: 165, blanc: 0, liberté de vote: 2, none: 11

PS gave a vote recommendation for 442 referenda
oui: 282, non: 138, blanc: 2, liberté de vote: 20, none: 13

UDC gave a vote recommendation for 442 referenda
oui: 224, non: 214, blanc: 0, liberté de vote: 4, none: 13

PLS gave a vote recommendation for 317 referenda
oui: 160, non: 154, blanc: 0, liberté de vote: 3, none: 17

AdI gave a vote recommendation for 233 referenda
oui: 153, non: 70, blanc: 1, liberté de vote: 9, none: 10

PEV gave a vote recommendation for 440 referenda
oui: 286, non: 147, blanc: 1, liberté de vote: 6, none: 15

PCS gave a vote recommendation for 180 referenda
oui: 104, non: 72, blanc: 0, liberté de vote: 4, none: 204

PST gave a vote recommendation for 411 referenda
oui: 213, non: 183, blanc: 0, liberté de vote: 15, none: 44

POCH gave a vote re

We can notice that parties are different from one another, bigger parties (UDC, PS, PRD, PDC) tend to support most referenda, whereas PLR, PVL, PBD don't. We can notice that Lega is the party that most frequently gave no recommendation on federal referenda, which can make sense as this party focuses mostly on its own canton (Ticino). "Liberté de vote" and "blanc" are rather rare, parties usually pick a side in referenda.

For the deeper data analysis and visualization, we have to make links between data. For instance, VERT-E-S is present there for only 2 referenda because it changed its name, it was formerly known as PES.